src/lib/y2storage/filesystems/btrfs.rb

```diff
@@ -0,0 +1,131 @@
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/storage_class_wrapper"
+require "y2storage/filesystems/blk_filesystem"
+require "y2storage/btrfs_subvolume"
+
+Yast.import "ProductFeatures"
+
+module Y2Storage
+  module Filesystems
+    # This is a wrapper for Storage::Btrfs
+    class Btrfs < BlkFilesystem
+      wrap_class Storage::Btrfs
+
+      # @!method top_level_btrfs_subvolume
+      #   Top level Btrfs subvolume
+      #
+      #   Btrfs filesystems always have a top level subvolume, the mkfs.btrfs
+      #   command implicitly creates it, so does libstorage when creating the
+      #   data structures.
+      #
+      #   The top level Btrfs subvolume always has ID 5.
+      #
+      #   @return [BtrfsSubvolume]
+      storage_forward :top_level_btrfs_subvolume, as: "BtrfsSubvolume"
+
+      # @!method default_btrfs_subvolume
+      #   The default Btrfs subvolume (typically @).
+      #
+      #   @note When a new Btrfs filesystem is created, a top level subvolume is added and
+      #   initialized as default subvolume.
+      #
+      #   @see #default_btrfs_subvolume
+      #
+      #   @return [BtrfsSubvolume]
+      storage_forward :default_btrfs_subvolume, as: "BtrfsSubvolume"
+
+      # @!method default_btrfs_subvolume=(subvolume)
+      #   Set the default Btrfs subvolume
+      #
+      #   @param subvolume [BtrfsSubvolume]
+      storage_forward :default_btrfs_subvolume=
+
+      # @!method btrfs_subvolumes
+      #   Collection of Btrfs subvolumes of the filesystem
+      #   @return [Array<BtrfsSubvolumes>]
+      storage_forward :btrfs_subvolumes, as: "BtrfsSubvolume"
+
+      # @!method find_btrfs_subvolume_by_path(path)
+      #   Finds a subvolume by its path
+      #
+      #   @param path [String] subvolume path
+      #   @return [BtrfsSubvolume, nil] nil if it does not find a subvolume with this path
+      storage_forward :find_btrfs_subvolume_by_path, as: "BtrfsSubvolume"
+
+      # Return the default subvolume, creating it when necessary
+      #
+      # If a specific default subvolume path is requested, create it; if not,
+      # use the toplevel subvolume that is implicitly created by mkfs.btrfs.
+      #
+      # This default subvolume is the parent for all others on Btrfs 'filesystem'.
+      #
+      # @param path [String, nil] path for the default subvolume
+      #
+      # @return [BtrfsSubvolume]
+      def get_or_create_default_btrfs_subvolume(path: nil)
```

Comment:
Only one detail. With that version, if default subvolume is "@":

```
filesystem.get_or_create_default_btrfs_subvolume(path: "")  # => subvolume for "@"
```
I guess it should be:

```
filesystem.get_or_create_default_btrfs_subvolume(path: "")  # => subvolume for ""
filesystem.get_or_create_default_btrfs_subvolume(path: nil)  # => subvolume for "@"
```